openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/RolesDetailPage/RolesDetailPage.tsx

```diff
@@ -0,0 +1,175 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { Card, Col, Empty, Row, Table, Tabs } from 'antd';
+import { ColumnsType } from 'antd/lib/table';
+import { AxiosError } from 'axios';
+import { isEmpty, uniqueId } from 'lodash';
+import { EntityReference } from 'Models';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { getRoleByName } from '../../../axiosAPIs/rolesAPIV1';
+import Description from '../../../components/common/description/Description';
+import RichTextEditorPreviewer from '../../../components/common/rich-text-editor/RichTextEditorPreviewer';
+import TitleBreadcrumb from '../../../components/common/title-breadcrumb/title-breadcrumb.component';
+import Loader from '../../../components/Loader/Loader';
+import {
+  GlobalSettingOptions,
+  GlobalSettingsMenuCategory,
+} from '../../../constants/globalSettings.constants';
+import { Role } from '../../../generated/entity/teams/role';
+import { getEntityName } from '../../../utils/CommonUtils';
+import {
+  getPolicyWithFqnPath,
+  getSettingPath,
+} from '../../../utils/RouterUtils';
+import { showErrorToast } from '../../../utils/ToastUtils';
+import './RolesDetail.less';
+
+const { TabPane } = Tabs;
+
+const RolesDetailPage = () => {
+  const { fqn } = useParams<{ fqn: string }>();
+
+  const [role, setRole] = useState<Role>({} as Role);
+  const [isLoading, setLoading] = useState<boolean>(false);
+
+  const breadcrumb = useMemo(
+    () => [
+      {
+        name: 'Roles',
+        url: getSettingPath(
+          GlobalSettingsMenuCategory.ACCESS,
+          GlobalSettingOptions.ROLES
+        ),
+      },
+      {
+        name: fqn,
+        url: '',
+      },
+    ],
+    [fqn]
+  );
+
+  const fetchRole = async () => {
+    setLoading(true);
+    try {
+      const data = await getRoleByName(fqn, 'policies,teams,users');
+      setRole(data ?? ({} as Role));
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const PoliciesList = ({ policies }: { policies: EntityReference[] }) => {
+    const columns: ColumnsType<EntityReference> = useMemo(() => {
+      return [
+        {
+          title: 'Name',
+          dataIndex: 'name',
+          width: '200px',
+          key: 'name',
+          render: (_, record) => (
+            <Link
+              className="hover:tw-underline tw-cursor-pointer"
+              to={getPolicyWithFqnPath(record.fullyQualifiedName || '')}>
+              {getEntityName(record)}
+            </Link>
+          ),
+        },
+        {
+          title: 'Description',
+          dataIndex: 'description',
+          key: 'description',
+          render: (_, record) => (
+            <RichTextEditorPreviewer markdown={record?.description || ''} />
```

Comment:
lets move this to separate component

Reply:
Same here, we are using a composition pattern here.